compiler/noirc_evaluator/src/brillig/brillig_ir/artifact.rs

```diff
@@ -11,7 +11,8 @@ pub(crate) enum BrilligParameter {
     /// An array parameter or return value. Holds the type of an array item and its size.
     Array(Vec<BrilligParameter>, usize),
     /// A slice parameter or return value. Holds the type of a slice item.
-    Slice(Vec<BrilligParameter>),
+    /// It can hold the slice size if known at compile time.
+    Slice(Vec<BrilligParameter>, Option<usize>),
```

Comment:
I don't see when this is ever `None`? In the case we call `ssa_type_to_parameter`? But we special handle slices in `gen_brillig_parameters` so the slice case should never be hit in `ssa_type_to_parameter`? Perhaps we should leave it to panic on `Type::Slice` and say slice types need to be handled with more than just type information and then we can make this a plain usize.